src/main/java/uk/gov/justice/hmpps/prison/service/transformers/OffenderTransformer.java

```diff
@@ -92,4 +97,14 @@ private InmateDetail.InmateDetailBuilder buildOffender(final Offender offender)
                 .ethnicity(offender.getEthnicity() != null ? offender.getEthnicity().getDescription() : null)
                 .build());
     }
+
+    public static List<OffenderSentenceTerms> filterSentenceTerms(List<SentenceTerm> terms, List<String> filterBySentenceTermCodes) {
+        final var sentenceTermCodes = (filterBySentenceTermCodes == null || filterBySentenceTermCodes.isEmpty()) ? List.of("IMP") : filterBySentenceTermCodes;
+        return terms
+            .stream()
+            .filter(term -> "A".equals(term.getOffenderSentence().getStatus()))
+            .filter(term -> sentenceTermCodes.contains(term.getSentenceTermCode()))
+            .map(SentenceTerm::getSentenceSummary)
```

Comment:
Wasn't sure if this was the best place for the method

Reply:
Feels like it should be in service - but you will move this to the HQL next anyway so DON'T WORRY ABOUT IT